### ISSUE TITLE
Local State is not working as Expected

### ISSUE THREAD

**Issue by thruthesky** [FlutterFlow/flutterflow-issues#272]
I know this is designed, not a bug. but I have to insist that the new `Update Local State` action is causing a real problem.
Please, correct if I am wrong.

## Expected behavior (required)
- When `Local State` changes, there should be a way of reflecting(rebuilding) only the necessary part.

## Current behavior (required)
- The new update of `Local State` has two option.
  1) Rebuild the whole screen
  2) Or don't rebuild anything.

## Context (required)

The new update does not solve this problem.

https://user-images.githubusercontent.com/5080329/209437715-90fe11a2-7747-461c-b702-e8f37dee9d46.mp4

I made it as a Component with custom action code.

Imagine that the app displays a carousel on each feed.
The developer made the carousel as a Component with some help of custom action.
When the user scroll down and see a feed with a carousel.
The user taps a thumbnail of the carousel, then the app should show the image of the thumbnail in big size.

To make this happen, the app needs to rebuild the Component only. And this is what `Local State` did before.

But now what `Local State` does is,
- it refresh the whole screen which break the screen by scrolling to the top of the screen,
- Or it does nothing which cause NO changes on the screen.

Thus, (I believe) this is not what FlutterFlow developers are looking for. And I think this is not only my issue.

We (developers) need a way to rebuild the Component only.

If `FFAppState.updateSilently(() { ... })` should go like
```
FFAppState.updateSilently(() {
  setState(() {
    FFAppState().varName = 'yo';
  });
});
```

Then, this would help.

The code above will still rebuild the whole screen if the `Update Local State` action happens on screen page which is controversial against the term `updateSilently`.
But if the `Update Local State` action happens on Component, it will rebuild the component only, not the whole active screen.

Or unless, if there is a `Update Widget State` that developer can declare a Widget State variable on each page, and update it just like `Upload Local State`.

Not all developers are good at building the Custom Widget to update only a part of the screen And building a custom widget is a waste when we can do it with a Component (if there is a way to rebuild the Component only).

What if the `Update Local State` action goes with a dropbox with options like

`Update Screen`
`Update Silently`
`Update Widget`

So, if the user choose `Update screen` (default), it goes `FFAppState.updateSilently(() { FFAppState.varName = ... })`
And if the user choose `Update Silently`, it goes `FFAppState().varName = ... `
And if the user choose `Update Widget`, it goes `FFAppState.updateSilently(() { setState(() { FFAppState.varName = ... }); });`

So, the developer has an option to rebuild the whole screen, or Component only, or nothing. And if the user choose `Update Widget State` on a page, then the page will be rebuilt.

**Comment by agreaves** [comment]
Hi @thruthesky, thank you very much for this report. This is something we have had in mind, and given your input and that of the rest of the community we've gone ahead and made this change! You can check it out now in your projects.

**Comment by thruthesky** [comment]
@agreaves Yes, I see it. Thank you very much :)

**Comment by Ellweb3** [comment]
Hi friends, @agreaves , @thruthesky how about if I need to update some standard text widgets or slider without redrawing the whole screen? How do I wrap them in setState without making it custom in code? Can we make an option for standard widgets - listen the state or it's better to set a default redraw of the widget if a variable from appState is specified in the value.
This is very important for my project.

**Comment by thruthesky** [comment]
Hi, @Ellweb3
Do you mean the widgets in Widget Trees of UI Builder? like Text, TextField, or Chart?
For Text widget,
1. you may make a component with the Text widget
2. update the text by setting `set from variable`.
3. And whenever you need to change the value(text) of the Text widget (in that component), You need to rebuild the component to reflect the changes. Then, use `Update Local State` with Component only rebuild option.


in summary, you need to rebuild the component to update(reset) the text of Text Widget. And to rebuild the component, use `Update Local State` with `Rebuild Component Only` option.

**Comment by Ellweb3** [comment]
@thruthesky  Thank you for your solution - it really will work but it's a crutch and might suggest the FF team to implement such a function to be able to change the state of non-custom widgets separately.

**Comment by OldFred2023** [comment]
is there a coding example of this?  is there a "updateSilently" method?

**Comment by OldFred2023** [comment]
> Hi @thruthesky, thank you very much for this report. This is something we have had in mind, and given your input and that of the rest of the community we've gone ahead and made this change! You can check it out now in your projects.

So is there a way to update App sate in a custom component without triggering the "context.watch<FFAppState>();" in other pages?
